woodwork/tests/test_utils.py

```diff
@@ -566,3 +569,452 @@ def test_parse_logical_type_errors():
+
+
+def test_concat_cols_with_duplicate_ww_indexes(sample_df):
+    df1 = sample_df[['id', 'phone_number', 'signup_date', 'email']]
+    df1.ww.init(index='id', time_index='signup_date')
+    df2 = sample_df[['full_name', 'age', 'id', 'signup_date', 'is_registered']]
+    df2.ww.init(index='id', time_index='signup_date')
+
+    combined_df = concat([df1, df2])
+    assert combined_df.ww.index == 'id'
+    assert combined_df.ww.time_index == 'signup_date'
+    assert len(set(combined_df.columns)) == len(set(sample_df.columns))
+
+    # confirm that the columns that are "lost" in the resulting df are still in the original dataframes
+    assert 'id' in df1.columns
+    assert 'signup_date' in df1.columns
+    assert 'id' in df2.columns
+    assert 'signup_date' in df2.columns
+
+
+def test_concat_rows_with_ww_indexes(sample_df):
+    sample_df.ww.init(index='id')
+    original_schema = sample_df.ww.schema
+    copy_df = sample_df.ww.copy()
+
+    if isinstance(sample_df, pd.DataFrame):
+        # Only pandas checks index uniqueness
+        error = 'Index column must be unique'
+        with pytest.raises(IndexError, match=error):
+            concat([sample_df, copy_df], axis=0)
+
+        # Without schema validation this does not get caught
+        combined_df = concat([sample_df, copy_df], axis=0, validate_schema=False)
+        assert combined_df.ww.index == 'id'
+
+    sample_df.ww.init(time_index='signup_date')
+    original_schema = sample_df.ww.schema
+    copy_df = sample_df.ww.copy()
+
+    combined_df = concat([sample_df, copy_df], axis=0)
+    assert len(combined_df) == 8
+
+    assert combined_df.ww.time_index == 'signup_date'
+    assert combined_df.ww.schema == original_schema
+
+
+def test_concat_table_names(sample_df):
+    df1 = sample_df[['id', 'signup_date']]
+    df1.ww.init(name=0)
+    df2 = sample_df[['full_name', 'age']]
+    df2.ww.init(name='1')
+    df3 = sample_df[['phone_number', 'is_registered']]
+    df3.ww.init(name='2')
+
+    combined_df = concat([df1, df2, df3])
+    assert combined_df.ww.name == '0_1_2'
+
+    combined_df = concat([df3, df2, df1])
+    assert combined_df.ww.name == '2_1_0'
+
+
+def test_concat_cols_different_use_standard_tags(sample_df):
+    sample_df.ww.init(use_standard_tags={'age': False, 'full_name': False, 'id': True},
+                      logical_types={'full_name': 'Categorical'})
+
+    assert sample_df.ww.semantic_tags['id'] == {'numeric'}
+    assert sample_df.ww.semantic_tags['full_name'] == set()
+    assert sample_df.ww.semantic_tags['age'] == set()
+
+    df1 = sample_df.ww[['id', 'full_name', 'email']]
+    df2 = sample_df.ww[['phone_number', 'age', 'signup_date', 'is_registered']]
+
+    combined_df = concat([df1, df2])
+
+    assert combined_df.ww.semantic_tags['id'] == {'numeric'}
+    assert combined_df.ww.semantic_tags['full_name'] == set()
+    assert combined_df.ww.semantic_tags['age'] == set()
+
+
+def test_concat_rows_different_schema(sample_df):
```

Comment:
It sounds like one way to simplify things is to be more restrictive about what inputs we allow. Ideally we wouldn't have to be so restrictive with axis=0 inputs that we'd require the table schemas to match exactly, but I do think it greatly simplifies things to require that table schemas are equal, though I'd suggest we do a shallow check because different metadata shouldn't stop us from being able to concat. 

I think that would also require all the DataFrames to have Woodwork initialized. Which is another requirement that would simplify things

Reply:
No longer allowing `axis=0`, so no longer a problem for this PR.